Framework/Algorithms/src/Stitch1DMany.cpp

```diff
@@ -76,6 +76,11 @@ void Stitch1DMany::init() {
       std::make_unique<VisibleWhenProperty>(useManualScaleFactorsTrue, manualScaleFactorsDefault, AND);
 
   setPropertySettings("ScaleFactorFromPeriod", std::move(scaleFactorFromPeriodVisible));
+
+  auto mustBePositive = std::make_shared<BoundedValidator<size_t>>();
+  mustBePositive->setLower(0);
+  declareProperty(std::make_unique<PropertyWithValue<size_t>>("IndexOfReference", 0, mustBePositive, Direction::Input),
```

Comment:
The algorithms can be called from python, which doesn't understand unsigned.
So size_t should be avoided in the properties. Use int but keep the validator.
Also for primitives, you can use the simplified override of declareProperty, i.e. you don't have to make an explicit PropertyWithValue<T>.